WebDriverAgentLib/Categories/XCUIElement+FBUtilities.m

```diff
@@ -52,35 +52,24 @@ - (BOOL)fb_waitUntilFrameIsStable
 
 - (XCElementSnapshot *)fb_lastSnapshot
 {
-  XCElementSnapshot *lastSnapshot = [self.fb_query fb_elementSnapshotForDebugDescription];
-  if (nil == lastSnapshot) {
-    [self fb_nativeResolve];
+  NSError *error = nil;
+  if (self.query.fb_isUniqueSnapshotSupported) {
+    self.lastSnapshot = [self.fb_query fb_uniqueSnapshotWithError:&error];
```

Comment:
I don't think it's a good idea to completely move to the uniqueSnapshot mechanism as the primary snapshot source if `fb_snapshotWithAttributes` fail since the resulting snapshot is only partial.

The attributes requested by uniqueSnapshot:
  "XC_kAXXCAttributeElementType",
  "XC_kAXXCAttributeElementBaseType",
  "XC_kAXXCAttributeParent",
  "XC_kAXXCAttributeIdentifier",
  "XC_kAXXCAttributeTraits"

The attributes requested by debugDescriptionSnapshot:
 "XC_kAXXCAttributeElementBaseType",
  "XC_kAXXCAttributeTruncatedValue",
  "XC_kAXXCAttributeViewControllerTitle",
  "XC_kAXXCAttributeUserTestingElements",
  "XC_kAXXCAttributeIdentifier",
  "XC_kAXXCAttributeTraits",
  "XC_kAXXCAttributeVerticalSizeClass",
  "XC_kAXXCAttributeFrame",
  "XC_kAXXCAttributeViewControllerClassName",
  "XC_kAXXCAttributeIsUserInteractionEnabled",
  "XC_kAXXCAttributeLabel",
  "XC_kAXXCAttributeURL",
  "XC_kAXXCAttributeParent",
  "XC_kAXXCAttributeHorizontalSizeClass",
  "XC_kAXXCAttributePlaceholderValue",
  "XC_kAXXCAttributeElementType"

While most of them are not needed, `XC_kAXXCAttributeTruncatedValue` and `XC_kAXXCAttributeFrame` are really important as they are used for element value and hitpoint calculation respectively.

In cases where `fb_snapshotWithAttributes` fails, this may cause incorrect snapshots to be returned.

Reply:
@Dan-Maor Could you please share your email address, so I could add you to Appium slack? I would like to discuss this topic in details.

Reply:
Sure, my Email is danmaor@outlook.com